templates/commands/render_action_append.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+
+	"github.com/abcxyz/abc/templates/model"
+)
+
+func actionAppend(ctx context.Context, ap *model.Append, sp *stepParams) error {
+	path, err := parseAndExecuteGoTmpl(ap.Path.Pos, ap.Path.Val, sp.inputs)
+	if err != nil {
+		return err
+	}
+
+	with, err := parseAndExecuteGoTmpl(ap.With.Pos, ap.With.Val, sp.inputs)
+	if err != nil {
+		return err
+	}
+
+	if err := walkAndModify(ctx, ap.Path.Pos, sp.fs, sp.scratchDir, path, func(buf []byte) ([]byte, error) {
+		// todo: should we add a newline before/after appending or leave that to user?
+		return append(buf, []byte(with)...), nil
```

Comment:
It's worth noting that we'll never iterate over the appended data. That's probably the _right_ decision, but consider:

1. An appended string that has the string "BANANA"
1. A regex replace rule for "BANANA"

If the actions are #1 and then #2, then the string will be replaced. However, if the actions are written as #2 and then #1, only instances of "BANANA" will be replaced in the original template, but the appended string will still contain "BANANA".

Again, I think this is probably the _right_ decision, but we should document this pretty heavily.

Reply:
As an aside, I wonder if we need debug functionality to print out the template after each action /cc @drevell

Reply:
I feel like this isn't unique to append. Pretty much any file modification will not be commutative. The same could be for two regex replaces in a row.

1. REGEX("FOO", "FOOBAR")
2. REGEX("BAR", "BARFOO")

If its 1 then 2, FOO->FOOBARFOO
If its 2 then 1, FOO->FOOBAR

I'm not certain extra documentation is needed besides general documentation that all actions are run in order.

Reply:
Filed #121 for improved template debugging support as suggested by Seth above.

Reply:
@sethvargo Do you still think we need documentation somewhere? I think its pretty intuitive that any mutations are done in order.

Reply:
I think we should document it, even if it's just in the README/examples. _Someone_ will open an issue, and it's always good to have something to point to.